File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  # given a string, the characters should flow from last to first
  # check for the index of the last leter of the word.
  # Will have to split the string into an array to access the index . str.split("")
  # move it to the front. 
  # then place the next letter after it
  # Move the rest of the characters to the next index
  # Do this for all the characters
  # Put them back together. 
  split_string = str.split("")
  # space = ''
  reversed = []
  str.size.times {reversed << split_string.pop} #then place the next letter after it
  reversed.join
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
  puts "Expecting: 'acirema si sith'"
  puts "=>", reverse_string('this is america')
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // Declare a variable to store the reversed value
  let reversedString = ''
  // Iterate through all the characters of the string, last to first
  for(let i= str.length - 1; i>=0; i--){
    reversedString += str[i]
  }
  return reversedString
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
